lib/active_merchant/billing/gateways/redsys.rb

```diff
@@ -485,7 +488,14 @@ def build_merchant_data(xml, data, options = {})
           xml.DS_MERCHANT_TERMINAL           options[:terminal] || @options[:terminal]
           xml.DS_MERCHANT_MERCHANTCODE       @options[:login]
           xml.DS_MERCHANT_MERCHANTSIGNATURE  build_signature(data) unless sha256_authentication?
-          xml.DS_MERCHANT_EXCEP_SCA          data[:sca_exemption] if data[:sca_exemption]
+
+          action = determine_3ds_action(data[:threeds]) if data[:threeds]
+          if action == 'iniciaPeticion' && data[:sca_exemption]
+            xml.DS_MERCHANT_EXCEP_SCA 'Y'
+          elsif data[:sca_exemption]
+            xml.DS_MERCHANT_EXCEP_SCA data[:sca_exemption]
+          end
+          xml.DS_MERCHANT_DIRECTPAYMENT data[:sca_exemption_direct_payment_enabled] if data[:sca_exemption_direct_payment_enabled]
```

Comment:
It seems that it will also be added for 'iniciaPeticion', but I think it was only needed for 'trataPeticion'.

Reply:
Good catch! will update